pkg/performance/collectors/process.go

```diff
@@ -0,0 +1,628 @@
+// Reference: https://www.kernel.org/doc/html/latest/filesystems/proc.html#process-specific-subdirectories
+type ProcessCollector struct {
+	performance.BaseContinuousCollector
+	ProcPath     string // Made public for testing
+	TopProcesses int    // Made public for testing
+	interval     time.Duration
+	procUtils    *procutils.ProcUtils
+
+	// State tracking for CPU percentage calculations
+	mu             sync.RWMutex
+	lastCPUTimes   map[int32]*ProcessCPUTime
+	lastUpdateTime time.Time
+}
+
+// ProcessCPUTime tracks CPU usage for a process over time
+type ProcessCPUTime struct {
+	TotalTime uint64
+	Timestamp time.Time
+}
+
+func NewProcessCollector(logger logr.Logger, config performance.CollectionConfig) (*ProcessCollector, error) {
+	// Validate paths are absolute
+	if !filepath.IsAbs(config.HostProcPath) {
+		return nil, fmt.Errorf("HostProcPath must be an absolute path, got: %q", config.HostProcPath)
+	}
+	if !filepath.IsAbs(config.HostSysPath) {
+		return nil, fmt.Errorf("HostSysPath must be an absolute path, got: %q", config.HostSysPath)
+	}
+
+	// Verify the proc path exists and is accessible
+	if info, err := os.Stat(config.HostProcPath); err != nil {
+		return nil, fmt.Errorf("HostProcPath not accessible: %w", err)
+	} else if !info.IsDir() {
+		return nil, fmt.Errorf("HostProcPath is not a directory: %q", config.HostProcPath)
+	}
+
+	capabilities := performance.CollectorCapabilities{
+		SupportsOneShot:    false,
+		SupportsContinuous: true,
+		RequiresRoot:       false,
+		RequiresEBPF:       false,
+		MinKernelVersion:   "2.6.0",
+	}
+
+	// Use configured value or default
+	topProcesses := config.TopProcessCount
+	if topProcesses <= 0 {
+		topProcesses = defaultTopProcessCount
+	}
+
+	interval := config.Interval
+	if interval <= 0 {
+		interval = 1 * time.Second
+	}
+
+	return &ProcessCollector{
+		BaseContinuousCollector: performance.NewBaseContinuousCollector(
+			performance.MetricTypeProcess,
+			"Process Statistics Collector",
+			logger,
+			config,
+			capabilities,
+		),
+		ProcPath:     config.HostProcPath,
+		TopProcesses: topProcesses,
+		interval:     interval,
+		lastCPUTimes: make(map[int32]*ProcessCPUTime),
+		procUtils:    procutils.New(config.HostProcPath),
+	}, nil
+}
+
+func (c *ProcessCollector) Start(ctx context.Context) (<-chan any, error) {
+	c.SetStatus(performance.CollectorStatusActive)
+
+	// Take initial snapshot of minimal stats to establish baseline
+	initial, err := c.collectMinimalStats(ctx)
+	if err != nil {
+		c.SetStatus(performance.CollectorStatusFailed)
+		return nil, fmt.Errorf("failed to collect initial process stats: %w", err)
+	}
+
+	c.mu.Lock()
+	c.updateLastCPUTimesFromMinimal(initial)
+	c.lastUpdateTime = time.Now()
+	c.mu.Unlock()
+
+	dataChan := make(chan any)
+	go c.runCollection(ctx, dataChan)
+	return dataChan, nil
+}
+
+func (c *ProcessCollector) Stop() error {
+	c.SetStatus(performance.CollectorStatusDisabled)
```

Comment:
So stop wouldn't close the data channel?

Perhaps we can orchestrate it like:
- Set status to disabled on context cancellation and calls to Stop which is what is currently done here
- But the `runCollection` goroutine is also listening for status updates and cleans up when status is disabled

I think what we really need to do is remove the context in Start because it makes the orchestration complicated with a separate Stop method but at least this way the resources get properly cleaned up.